scripts/cc-cli/src/test/integration-tests/collectCoins.test.ts

```diff
@@ -0,0 +1,138 @@
+  KeyringPair,
+  Wallet,
+  creditcoinApi,
+  providers,
+} from "creditcoin-js";
+import { cryptoWaitReady } from "@polkadot/util-crypto";
+import { testData } from "creditcoin-js/lib/testUtils";
+import { describeIf } from "../../utils/tests";
+
+describeIf(arg("CREDITCOIN_EXECUTE_SETUP_AUTHORITY"), "collect-coins", () => {
+  let ccApi: CreditcoinApi;
+  let sudo: KeyringPair;
+  let caller: any;
+
+  const { keyring, blockchain } = testData(
+    arg("CREDITCOIN_ETHEREUM_CHAIN") as Blockchain,
+    arg("CREDITCOIN_CREATE_WALLET"),
+  );
+
+  beforeAll(async () => {
+    await deployCtcToken((global as any).CREDITCOIN_CTC_CONTRACT_ADDRESS);
+    setArg(
+      "CREDITCOIN_CTC_CONTRACT_ADDRESS",
+      process.env.CREDITCOIN_CTC_CONTRACT_ADDRESS,
+    );
+    setArgIfUndef(
+      "CREDITCOIN_CTC_BURN_TX_HASH",
+      process.env.CREDITCOIN_CTC_BURN_TX_HASH,
+    );
+
+    await retry(5, async (resolve: any, reject: any) => {
+      await setAuthorities().then(resolve).catch(reject);
+    }).catch(() => {
+      console.log("Could not setup testing authorities");
+      process.exit(1);
+    });
+
+    await cryptoWaitReady();
+
+    caller = randomTestAccount(false);
+
+    ccApi = await creditcoinApi((global as any).CREDITCOIN_API_URL);
+    sudo = arg("CREDITCOIN_CREATE_SIGNER")(keyring, "lender");
+
+    const { api } = ccApi;
+
+    /* eslint-disable @typescript-eslint/naming-convention */
+    const contract = api.createType(
+      "PalletCreditcoinOcwTasksCollectCoinsDeployedContract",
+      {
+        address: (global as any).CREDITCOIN_CTC_CONTRACT_ADDRESS,
+        chain: blockchain,
+      },
+    );
+
+    await api.tx.sudo
+      .sudo(api.tx.creditcoin.setCollectCoinsContract(contract))
+      .signAndSend(sudo, { nonce: -1 });
+  }, 100_000);
+
+  afterAll(async () => {
+    await ccApi.api.disconnect();
+  });
+
+  test("e2e", async () => {
+    const { api } = ccApi;
+
+    const provider = new providers.JsonRpcProvider(
+      arg("CREDITCOIN_ETHEREUM_NODE_URL"),
+    );
+    const deployerWallet = new Wallet(
+      arg("CREDITCOIN_CTC_DEPLOYER_PRIVATE_KEY"),
+      provider,
+    );
+
+    const fundTx = await fundFromSudo(
+      caller.address,
+      parseAmountInternal("1000000"),
+      arg("CREDITCOIN_API_URL"),
+    );
+    await signSendAndWatch(fundTx, api, sudo);
+
+    const registerResult = execa.commandSync(
+      `npx creditcoin-cli register-address -u ${
+        arg("CREDITCOIN_API_URL") as string
+      }`,
+      {
+        env: {
+          BLOCKCHAIN: "Ethereum",
+          PRIVATE_KEY: arg("CREDITCOIN_CTC_DEPLOYER_PRIVATE_KEY"),
+          CC_SECRET: caller.secret,
+        },
+      },
+    );
+
+    const stdout = registerResult.stdout.split("\n");
+    expect(registerResult.failed).toBe(false);
+    expect(registerResult.exitCode).toBe(0);
+    expect(registerResult.stderr).toBe("");
+    expect(
+      stdout[stdout.length - 1].includes("Address Registered Successfully"),
+    ).toBe(true);
```

Comment:
Since this is `collectCoins` you don't really need to assert on the output of register-address here (that's done explicitly in `registerAddress.test.ts` anyway. 

You can also go ahead and use `tryRegisterAddress` from creditcoin-js